memory/table.go

```diff
@@ -123,12 +124,12 @@ var _ sql.PrimaryKeyAlterableTable = (*Table)(nil)
 var _ sql.PrimaryKeyTable = (*Table)(nil)
 
 // NewTable creates a new Table with the given name and schema.
-func NewTable(name string, schema sql.PrimaryKeySchema, fkColl *ForeignKeyCollection) *Table {
-	return NewPartitionedTable(name, schema, fkColl, 0)
+func NewTable(name string, schema sql.PrimaryKeySchema, fkColl *ForeignKeyCollection, collation sql.CollationID) *Table {
```

Comment:
Should avoid this change if possible. We have people using these constructors in tests that will be broken by this change.

Introduce a new method NewTableWithCollation, or a builder pattern, or something that avoids having to update every call site as this PR does.

Reply:
Makes sense, I'll add a new function for it